fitwsarah_frontend/src/views/AdminPanelPage/AdminAppointments.js

```diff
@@ -0,0 +1,130 @@
+    } = useAuth0();
+
+    const [appointments, setAppointments] = useState([]);
+    const [accessToken, setAccessToken] = useState(null);
+
+    useEffect(() => {
+        if (isAuthenticated) {
+            const getAccessToken = async () => {
+                try {
+                    const token = await getAccessTokenSilently({
+                        audience: configData.audience,
+                        scope: configData.scope,
+                    });
+                    setAccessToken(token);
+                } catch (e) {
+                    console.error(e.message);
+                }
+            };
+            getAccessToken();
+        }
+    }, [getAccessTokenSilently, isAuthenticated]);
+
+    useEffect(() => {
+        if (accessToken) {
+            getAllAppointments();
+        }
+    }, [accessToken]);
+
+
+
+
+    const getAllAppointments = () => {
+        fetch("http://localhost:8080/api/v1/appointments", {
+            method: "GET",
+            headers: new Headers({
+                Authorization: "Bearer " + accessToken,
+                "Content-Type": "application/json"
+            })
+        })
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok ' + response.statusText);
+                }
+                return response.json();
+            })
+            .then((data) => {
+                console.log(data);
+                setAppointments(data);
+            })
+            .catch((error) => {
+                console.log(error);
+
+            });
+    };
+
+
+    return (
+        <div>
+            {!isAuthenticated && <NavNotLoggedIn/>}
+            {isAuthenticated && <NavLoggedIn/>}
+            <div className="accounts-section">
+                <div className="container">
+                    <Link to="/adminPanel" className="button back-button">Back</Link>
+                    <h1>Appointments</h1>
+                    <input
+                        type="text"
+                        className="search-bar"
+                        placeholder="Search..."
+                    />
+                    <FaSearch className="search-icon" />
+                    <div className="table-responsive">
+                        <table className="table">
+                            <thead>
+                            <tr>
+                                <th>Appointment Id</th>
+                                <th>Account Id</th>
+                                <th>Availability Id</th>
+                                <th>Admin Id</th>
+                                <th>Service Id</th>
+                                <th>Status</th>
+                                <th>Location</th>
+                            </tr>
+                            </thead>
+                            <tbody>
+                            {appointments.map(appointments => (
+                                <tr key={appointments.id}>
```

Comment:
What does this do?

Reply:
This essentially maps every appointment in the body to their respective "appointmentId".